blueprints/hr_assistant/general.py

```diff
@@ -0,0 +1,640 @@
+    # for numerical entities using the request object
+    if not num_entity:
+        num_entity = [float(e['value'][0]['value'])
+                      for e in request.entities
+                      if e['type'] == 'sys_number']
+
+        for i in request.text.split():
+            try:
+                num_entity.append(float(i))
+            except Exception:
+                continue
+
+        # keep unique
+        num_entity = list(set(i for i in num_entity))
+
+    extreme_canonical = extreme_entity['value'][0]['cname']
+
+    if extreme_canonical == 'highest':
+        qa = qa.sort(field=field, sort_type='desc')
+
+    elif extreme_canonical == 'lowest':
+        qa = qa.sort(field=field, sort_type='asc')
+
+    # Respond according to the number of extreme cases asked for
+    # else default to a single response
+
+    # Eg. 'who are the 5 youngest employees'
+    if num_entity and num_entity[0] <= SIZE:
+        size = num_entity[0]
+
+    # Eg. 'who is the youngest employee'
+    else:
+        size = 1
+
+    return qa, size
+
+
+def _agg_function(qa_out, func='avg', num_col='money'):
+    """
+    Function that does Sum, Average and Percent Calculations
+    param qa_out (list) List of Json Objects Representing Users
+    param func (str) - Function Type: 'avg','sum', 'ct', 'pct'
+    param num_col (str) - Numerical Column Type : 'money', or 'age'
+    returns result (float) - Resulting Value from function operation
+    """
+
+    try:
+        if(func == 'avg'):
+            return round(np.mean([emp[num_col] for emp in qa_out]), 2)
+        elif(func == 'sum'):
+            return np.sum([emp[num_col] for emp in qa_out])
+        elif(func == 'ct'):
+            return len(qa_out)
+        elif(func == 'pct'):
+            return round(len(qa_out)/3, 2)
+    except Exception:
+        return 0
+
+
+def _get_names(qa_out):
+    """
+    Get a List of Names from a QA Result
+    param qa_out (list) Output of QA from a query
+    """
+
+    names = [str(out['first_name']) + ' ' + str(out['last_name']) for out in qa_out]
+    names = ', '.join(names)
+    return names
+
+
+def _get_person_info(request, responder, entity_type):
+    """
+    This function is used to fetch employee names from either the previous turns' context
+    or the current query. The latest occurence of a name entity is given preference.
+    The name is then passed to the _fetch_from_kb function to obtain the desired
+    information about the employee under consideration from the Knowledge Base.
+    """
+
+    name = request.frame.get('name')
+
+    # if the user has provided a new name, replace the existing name with it
+    try:
+        name_ent = [e for e in request.entities if e['type'] == 'name']
+        name = name_ent[0]['value'][0]['cname']
+    except Exception:
+        if name:
+            pass
+        else:
+            return responder
+
+    # If name in database, fetch details from knowledge base
+    if name != '':
```

Comment:
this should be sufficient

```suggestion
    if name'':
```